### ISSUE TITLE
TypeScript 5.6 causes NG3 error

### ISSUE THREAD

**Issue by e-oz** [angular/angular#57158]
### Which @angular/* package(s) are the source of the bug?

compiler

### Is this a regression?

No

### Description

I know that `@angular/compiler` does not support TS 5.6 yet, just a heads-up:

https://stackblitz.com/~/github.com/e-oz/stackblitz-starters-wchtmf?file=src/main.ts

### Please provide a link to a minimal reproduction of the bug

https://stackblitz.com/~/github.com/e-oz/stackblitz-starters-wchtmf?file=src/main.ts

### Please provide the exception or error you saw

```true
"This kind of expression is always falsy."
```

### Please provide the environment you discovered this bug in (run `ng version`)

```true
Angular 18.2-next
TypeScript 5.6.0-beta
```

**Comment by JeanMeche** [comment]
TS 5.6 is not supported yet.

**Comment by JeanMeche** [comment]
This also raises an error on previous versions right ?

**Comment by JoostK** [comment]
This is a new error due to new diagnostics being introduced in TS 5.6. These will be addressed when the repo as updated to TS 5.6, as that will expose these errors from the type-checking tests.
